packages/relay/src/services/UserService.ts

```diff
@@ -90,6 +86,11 @@ export class UserService {
         fromServer: boolean,
         synchronizer: UnirepSocialSynchronizer
     ) {
+        // verify attesterId, should be the same as app
+        const clientAttesterId = this.getAttesterId(publicSignals[2])
+        if (synchronizer.attesterId != clientAttesterId)
+            throw new Error('Invalid attesterId')
+
```

Comment:
there is a decoded signal in `signupProof` so you can simply use
```ts
   if (synchronizer.attesterId != signupProof.attesterId)
            throw new Error('Invalid attesterId')
```
https://developer.unirep.io/docs/circuits-api/signup-proof#attesterid

If it is wrong please report to unirep repo 🙏🏻

Reply:
the `private getAttesterId(control?: string)` can be removed as well,
you can use the `signupProof` in 
https://github.com/social-tw/social-tw-website/blob/dcf3a18c2110f5df703db64f1e7f7bdf29ed1501/packages/relay/src/services/UserService.ts#L94-L98
and check attester id after `signupProof` is constructed
```ts
if (synchronizer.attesterId != signupProof.attesterId)
            throw new Error('Invalid attesterId')
```